broker/tasks/cloudfront.py

```diff
@@ -71,6 +71,18 @@ def get_custom_error_responses(service_instance):
         return {"Quantity": 0}
 
 
+def is_cdn_with_dedicated_waf_instance(service_instance) -> bool:
+    return (
+        isinstance(service_instance, CDNDedicatedWAFServiceInstance)
+        and service_instance.dedicated_waf_web_acl_arn
+    )
+
+
+def update_cdn_with_dedicated_waf_instance_tags(tags: list[Tag]) -> list[Tag]:
```

Comment:
(opinion) this feels a little _too_ DRY, at least as a bare function. Should it be an instance method instead?